### PR TITLE
Remove Winter GM Page

### PR DESCRIPTION
**Pull request by advaithg**
Addresses issue #123

Changes made:
1. Removed references to GM from navbar
2. Replaced most common data using placeholders - placed data in **config/gmData.js**
3. Imported data file and **replaced hardcoded information** with placeholders
4. Added **comments** detailing gmData file and how to add back to navbar

Did not remove route from router - page is still accessible through '**/gm** '

### CODE REVIEW

File: src/components/Events/GM/GM.js
Comment (evanzhong) [inline src/components/Events/GM/GM.js:98]: you can similarly try string templating here

File: src/components/Events/GM/GM.js
Comment (evanzhong) [inline src/components/Events/GM/GM.js:129]: I would actually remove this line entirely, or generalize it rather than commenting it out

File: src/components/Events/GM/GM.js
Comment (evanzhong) [inline src/components/Events/GM/GM.js:164]: Not a huge issue, but you can consider moving the function call to `toLocaleString` into a separate helper function. (i.e. this bit `GM_START_TIME.toLocaleString('en-US', {weekday: 'long', year: 'numeric', month: 'long', day: 'numeric' })`)

In my personal experience, moving as much complicated logic (or javascript expressions/statements that take a moment to read) to dedicated functions helps with code separation. 

In brief, you want to keep as much of our "template" (html) separate from our "controlling logic" that actually generates the dynamic content.

File: src/config/gmData.js
Comment (evanzhong) [inline src/config/gmData.js:1]: Over data structure looks great!

I'm now realizing that these data files would be great places for Typescript, so we can enforce more strict compile time rules about formatting (i.e. the comment you wrote below about `month` and `day`. Would love to hear @mattxwang's take on this.

Perhaps out of scope for this PR, but you could also consider adding some smoke tests for the data. (i.e. regex checking length of strings, formatting etc).

File: src/config/gmData.js
Comment (evanzhong) [inline src/config/gmData.js:109]: I'm not sure what our exact procedure for our codebase is, but my preference is to define each data structure as just a constant variable, then consolidating all export statements at the end.

Something along the lines of:
```suggestion
];

export {
    boardEvents,
    committeeData,
    //your default obj
}
```

Alternatively you could just wrap everything as one big ol' object, then export the large wrapper object as a default.

**Comment by mattxwang** [comment]
As a quick comment, I just merged #147 which slightly changes the `eslint` behaviour (it does an extra autofix step), when you have a chance @advaithg can you merge from `master` and rerun + push?

File: src/config/gmData.js
Comment (advaithg) [reply src/config/gmData.js:1]: I ended up deciding against that since I wasn't sure what the error message or error handling situation for that should be - since right now giving it an invalid input makes date NAN - which makes it obvious along with the comment what the error is.

Do you think I should still do it? If so what kind of error message would you say make sense?

Comment (evanzhong) [reply src/config/gmData.js]: I think it's probably more work than its worth and I'd rather ship this PR as is for the following reasons
- The data model is pretty well documented in comments and in this PR
- This file will likely see pretty light use, given we only have 2 GMs every year

Final call is up to you.

Comment (advaithg) [reply src/config/gmData.js]: This makes sense! 
Once it's done being reviewed I'll merge it into main then.

**Review by evanzhong — APPROVED** [review]
lgtm!

**Review by annaguo1012 — APPROVED** [review]
LGTM! I think Evan covered everything. I really like how this turned out and appreciate the helpful comments for future devs! :)

**Pull request merged**
